### PR TITLE
Test improvements (part 2)

### PR DESCRIPTION
**Pull request by eliasnogueira**
These are small improvements in some existing tests plus the creation of an additional to exercise the corner cases of the Pagination class.
A test guideline was also included to give contributors an idea about how the project can be tested.

## Added

* Added testing guideline
* Added `PaginationTest`

## Changed

* Removed `public` keyword in the `KeysetPageableTest` and `PageableTest` methods
* Replaced JUnit 5 assertions to the AssertJ in the `PageableTest`
* Bump AssertJ version

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-= Changelog
+w= Changelog
```
Comment (otaviojava) [inline CHANGELOG.adoc:1]: Was this `w` an accident?

Comment (eliasnogueira) [reply CHANGELOG.adoc]: it was. My bad.

It's now fixed.

**Review by otaviojava — APPROVED** [review]

**Comment by otaviojava** [comment]
Approved!

**Comment by otaviojava** [comment]
btw @eliasnogueira what do you think if you move it to the https://github.com/eclipse/jnosql
You don't need to fix all, but with this guide, we can create a task team to improve the test issues.
